packages/plugins/response-cache/src/plugin.ts

```diff
@@ -304,22 +288,73 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
           const schemaCoordinates = `${typeName}.${fieldName}`;
           const resultTypeNames = unwrapTypenames(fieldConfig.type);
           typePerSchemaCoordinateMap.set(schemaCoordinates, resultTypeNames);
+
+          if (idFields.includes(fieldName) && !idFieldByTypeName.has(typeName)) {
+            idFieldByTypeName.set(typeName, fieldName);
+          }
+
           if (cacheControlDirective) {
             const cacheControlAnnotations = getDirective(schema, fieldConfig, 'cacheControl');
             cacheControlAnnotations?.forEach(cacheControl => {
               const ttl = cacheControl.maxAge * 1000;
               if (ttl != null) {
-                ttlPerSchemaCoordinate[`${typeName}.${fieldName}`] = ttl;
+                ttlPerSchemaCoordinate[schemaCoordinates] = ttl;
               }
               if (cacheControl.scope) {
-                scopePerSchemaCoordinate[`${typeName}.${fieldName}`] = cacheControl.scope;
+                scopePerSchemaCoordinate[schemaCoordinates] = cacheControl.scope;
               }
             });
           }
           return fieldConfig;
         },
       });
-      processedSchemas.add(schema);
+
+      const typeInfo = new TypeInfo(schema);
```

Comment:
In theorie, it is reusable because once you visited all the tree, we come back to the initial empty state. But It's probably not a good idea to rely on the fact the visitor actually traversed the whole tree :-(

This means we have to instantiate a new visitor for each execution :-/

Reply:
Reverting this logic;
https://github.com/n1ru4l/envelop/pull/1970/files#diff-cf77bd9608a2b0bf192f49649a89018e1c8d8d171b0ee9f7d3f5eca4e230d656R211
and moving TypeInfo instantiation to the function body should be ok. But in this case, memoization should contain schema as well.